p2pool/src/sharechain/p2chain.rs

```diff
@@ -231,6 +242,15 @@ impl<T: BlockCache> P2Chain<T> {
                 warn!(target: LOG_TARGET, "Block squad mismatch, skipping block");
                 continue;
             }
+            if !block.verified.is_verified() {
+                warn!(target: LOG_TARGET, "Block not verified, skipping block");
+                continue;
+            }
+
+            if block.timestamp > earliest_date {
```

Comment:
Logic is inverted here
```suggestion
            if block.timestamp < earliest_date {
```

Reply:
yeah